src/Tuttifrutti/ServantClient.hs

```diff
@@ -0,0 +1,55 @@
+-- | Wrappers for Servant.Client
+--
+--   We need to wrap some Servant.Client functions so we can
+--   log requests and pass the X-Request-Id header forward.
+
+module Tuttifrutti.ServantClient
+  ( createLoggingManager
+  , wrapClientEnv
+  ) where
+
+import Tuttifrutti.Prelude
+
+import           Data.Aeson.Key         (toText)
+import           Servant.Client         (ClientEnv(..))
+import           Servant.Client.Core    (addHeader)
+import           Tuttifrutti.Log        as Log
+import           Tuttifrutti.Log.Handle as Log
+import qualified Data.Aeson as Aeson
+import qualified Tuttifrutti.Http       as Http
+import qualified Tuttifrutti.RequestId  as RequestId
+
+-- | Create a manager which logs all requests using the specified logger,
+--   adding request_id to the log message from the request headers.
+--   msg should be eg. "Lettera -> Persona request"
+createLoggingManager :: MonadIO m => Text -> Log.Handle -> Http.ManagerSettings -> m Http.Manager
+createLoggingManager msg logHandle settings =
+  Http.newManagerSettings $ settings
+          { Http.managerModifyRequest = \r -> do
+                -- it would be super nice to grab the request id from anything else,
+                -- but this is all we have in this context
+                let requestId = decodeUtf8Lenient . snd <$> find (\(h, _) -> h == "X-Request-Id") (Http.requestHeaders r)
+                with logHandle $
+                    Log.logInfo msg
+                       [ "url" .= decodeUtf8Lenient (Http.path r)
+                       , "method" .= decodeUtf8Lenient (Http.method r)
+                       , "request_id" .= fromMaybe "[none]" requestId
+                       ]
+                Http.managerModifyRequest Http.defaultManagerSettings r
+          }
+
+unwrapAesonString :: Aeson.Value -> Maybe Text
+unwrapAesonString (Aeson.String t) = Just t
+unwrapAesonString _ = Nothing
+
+-- | Wrap a ClientEnv so that it adds the X-Request-ID header based on
+--   the current requestId in the logger
+wrapClientEnv :: MonadIO m => ClientEnv -> Log.Handle -> m ClientEnv
+wrapClientEnv ClientEnv {..} logger = do
```

Comment:
I'm surprised if this is valid syntax. Remove the space in between  `ClientEnv {..}`, spaces in general (unless overridden with brackets) in function definitions separate arguments and on a careless read this reads like a 3 parameter function.